### ISSUE TITLE
Adding train sample code to detection models

### ISSUE THREAD

**Issue by prajjwal1** [pytorch/vision#1688]
This is an addition to existing documentation provided by torchvision docs. 
For instance, only the inference sample code has been provided
```
>>> model.eval()
>>> x = [torch.rand(3, 300, 400), torch.rand(3, 500, 400)]
>>> predictions = model(x)
```
It would be better if the docs also contains sample code for what model would expect in train mode, for example, 
```
images = torch.rand(2, 3, 600, 1200)
images = list(image for image in images)

boxes = torch.rand(2,11,4)
labels = torch.rand(2,11)

targets = []
for i in range(len(images)):
    d = {}
    d['boxes'] = boxes[i]
    d['labels'] = labels[i].type(torch.int64)
    targets.append(d)
    d = {}

model = fasterrcnn_resnet50_fpn(pretrained=False, progress=True,
                            num_classes=91, pretrained_backbone=True)

model(images,targets)
```
This would give the user a decent idea what `GeneralizedRCNNTransform` would expect in case of batch size>1. 

@fmassa I can open a PR for this. I think it would be useful to have this information.

**Comment by fmassa** [comment]
Sure, improving the documentation would be great!

Maybe pointing out to the tutorial in https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html would also be helpful?

**Comment by prajjwal1** [comment]
Yes, should I add the proposes change to the docstrings of the code [here](https://github.com/pytorch/vision/blob/master/torchvision/models/detection/faster_rcnn.py) or just make changes to the tutorial or both ?

**Comment by fmassa** [comment]
Improving the docstring adding a few links to the tutorials would be great!

**Comment by prajjwal1** [comment]
I've opened PRs in [vision](https://github.com/pytorch/vision/pull/1695) and [tutorials](https://github.com/pytorch/tutorials/pull/796). Please review them and let me know, if any modification is required.

**Comment by prajjwal1** [comment]
For some reason, some checks are not passing. What can I do to fix it ?

**Comment by fmassa** [comment]
@prajjwal1 I fixed the tests in a follow-up PR (lint was failing), but can you do the same for the tutorials? I've just commented in the PR

**Comment by prajjwal1** [comment]
I've done it. Please see the updated PR.

**Comment by fmassa** [comment]
@prajjwal1 I've approved it, but I don't have merge rights to that repository

**Comment by prajjwal1** [comment]
Closing this, both the PRs got merged. Thanks @fmassa for your support.
